server/api-manager.js

```diff
@@ -35,32 +39,895 @@ class MindbodyQueries {
+      URL_SITE + "memberships" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Optional:
+  *  @param ScheduleType
+  *  @param OnlineOnly
+  *//*eslint-enable */
+  getPrograms(parameters) {
+    var query = QueryString.stringify(parameters);
+    var request = new MindbodyRequest(
+      URL_SITE + "programs" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Optional:
+  *  @param SessionTypeIds
+  *  @param LocationId
+  *  @param StartDateTime 
+  *  @param EndDateTime 
+  *//*eslint-enable */
+  getResources(parameters) {
+    var query = QueryString.stringify(parameters);
+    var request = new MindbodyRequest(
+      URL_SITE + "resources" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Optional:
+  *  @param ProgramIds
+  *  @param OnlineOnly
+  *//*eslint-enable */
+  getSessionTypes(parameters) {
+    var query = QueryString.stringify(parameters);
+    var request = new MindbodyRequest(
+      URL_SITE + "sessiontypes" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Optional:
+  *  @param SiteIds 
+  *//*eslint-enable */
+  getSites(parameters) {
+    var query = QueryString.stringify(parameters);
+    var request = new MindbodyRequest(
+      URL_SITE + "sites" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Required:
+  *  @param SessionTypeId 
+  * Optional:
+  *  @param Filters  
+  *  @param StaffIds  
+  *  @param StaffIds  
+  *  @param LocationId  
+  *//*eslint-enable */
+  getStaff(parameters) {
+    var query = QueryString.stringify(parameters);
+    var request = new MindbodyRequest(
+      URL_STAFF + "staff" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Required:
+  *  @param StaffId 
+  *//*eslint-enable */
+  getStaffPermissions(parameters) {
+    var query = QueryString.stringify(parameters);
+    var request = new MindbodyRequest(
+      URL_STAFF + "staffpermissions" + query,
+      APIKEY,
+      SITEID,
+      "GET",
+      ""
+    );
+    return this.decorateAndMake(request);
+  }
+
+
+
+  //@TODO implement caching based on hashes of url's here
+  decorateAndMake(request) {
+    request.addAuth(this.authToken);
     this.requestNum++;
     if (!this.atLimit()) {
-      return req.makeRequest();
+      return request.makeRequest();
     }
     return Promise.reject(Error("Request limit reached"));
   }
 
+
   //may be changed later
   atLimit() {
+    /*eslint-disable no-magic-numbers */
```

Comment:
I think it's better to give eslist what it wants here and write like:
    const MAX_REQUESTS_DESIRED = 800;
    if (this.requestNum >= MAX_REQUESTS_DESIRED) {
      return true;

Reply:
Agreed, btw you can suggest changes with a dropdown when you are in review mode, which lets the owner push a button to merge the change.